lib/functions/image/partitioning.sh

```diff
@@ -214,17 +214,39 @@ function prepare_partitions() {
 						echo "$bootpart : name=\"bootfs\", start=${next}MiB, size=${BOOTSIZE}MiB, type=${type}"
 						local next=$(($next + $BOOTSIZE))
 					else
-						# no `size` argument mean "as much as possible"
+						# No 'size' argument means "expand as much as possible"
 						echo "$bootpart : name=\"bootfs\", start=${next}MiB, type=${type}"
 					fi
 				fi
 
 				# Root filesystem partition
 				if [[ -n "$rootpart" ]]; then
 					# dos: Linux
-					# gpt: Linux filesystem
-					[[ "$IMAGE_PARTITION_TABLE" != "gpt" ]] && local type="83" || local type="0FC63DAF-8483-4772-8E79-3D69D8477DE4"
-					# no `size` argument mean "as much as possible"
+					# gpt: Linux root
+					if [[ "$IMAGE_PARTITION_TABLE" != "gpt" ]]; then
+						local type="83"
+					else
+						# Linux root has a different Type-UUID for every architecture
+						# See https://uapi-group.org/specifications/specs/discoverable_partitions_specification/
+						case "$ARCH" in
```

Comment:
I'd introduce a variable (say `ARCH_ROOTFS_GPT_TYPE` or such) in each arch file in `config/sources`, and default to previous value if unset. That way we avoid a switch/case as well.